waspc/src/Wasp/Lib.hs

```diff
@@ -43,75 +45,84 @@ compile ::
   CompileOptions ->
   IO ([CompileWarning], [CompileError])
 compile waspDir outDir options = do
-  (analyzerWarnings, appSpecOrAnalyzerErrors) <- analyzeWaspProject waspDir options
-  compilerWarningsAndErrors <- case appSpecOrAnalyzerErrors of
-    Left analyzerErrors -> return ([], toList analyzerErrors)
-    Right appSpec ->
-      case ASV.validateAppSpec appSpec of
-        [] -> do
-          (generatorWarnings, generatorErrors) <- Generator.writeWebAppCode appSpec outDir (sendMessage options)
-          return (map show $ generatorWarningsFilter options generatorWarnings, map show generatorErrors)
-        validationErrors -> do
-          return ([], map show validationErrors)
-  return $ (analyzerWarnings, []) <> compilerWarningsAndErrors
-
-analyzeWaspProject ::
+  compileWarnings <- maybeToList <$> warnIfDotEnvPresent waspDir
+  appSpecOrCompileErrors <- analyzeProject waspDir options
+  compileWarningsAndErrors <- case appSpecOrCompileErrors of
+    Left analyzerErrors -> return (compileWarnings, analyzerErrors)
+    Right appSpec -> generateCode appSpec outDir options
+  return $ (compileWarnings, []) <> compileWarningsAndErrors
```

Comment:
Suggestion with couple improvements:
 - possible naming improvement
 - a bug fix (you were adding compileWarnings twice into the mix -> if there were analyzer errors, you would end up with them being repeated twice)
 - used lambda case to avoid necessarily capturing and naming output of analyzer -> I never used it like this but I google a bit and it is really cool usage of lambda case!

```suggestion
  compileWarningsAndErrors <-
    analyzeProject waspDir options >>= \case 
      Left analyzerErrors -> return ([], analyzerErrors)
      Right appSpec -> generateCode appSpec outDir options
  dotEnvWarnings <- maybeToList <$> warnIfDotEnvPresent waspDir
  return $ (dotEnvWarnings, []) <> compileWarningsAndErrors
```